app/code/core/Mage/Catalog/Model/Product/Image.php

```diff
@@ -244,84 +244,6 @@ public function setSize($size)
         return $this;
     }
 
-    /**
-     * @param null $file
-     * @return bool
-     */
-    protected function _checkMemory($file = null)
```

Comment:
also removing the function may break modules using it. Please only remove the usages, and optionally mark the method as deprecated.
Please also do not remove the functionality, yet.